hbase-server/src/main/java/org/apache/hadoop/hbase/master/normalizer/RegionNormalizerWorker.java

```diff
@@ -207,14 +214,34 @@ private List<NormalizationPlan> calculatePlans(final TableName tableName) {
       return Collections.emptyList();
     }
 
-    final List<NormalizationPlan> plans = regionNormalizer.computePlansForTable(tblDesc);
+    List<NormalizationPlan> plans = regionNormalizer.computePlansForTable(tblDesc);
+
+    plans = truncateForSize(plans);
+
     if (CollectionUtils.isEmpty(plans)) {
       LOG.debug("No normalization required for table {}.", tableName);
       return Collections.emptyList();
     }
     return plans;
   }
 
+  private List<NormalizationPlan> truncateForSize(List<NormalizationPlan> plans) {
```

Comment:
Thanks for moving. One more request here -- please add a LOG.debug when limit is reached. We should at least include the original number of plans, number of plans that will be submitted, and cumulativeSizeMb (maybe the plan that tipped it over the limit?).  This will help operators diagnose issues where the normalizer seems to be doing less work than they'd expect.  It might be nice to include the total size, which I'm realizing might help aid in tuning but of course that means removing the loop break I asked you to make.

Reply:
Done

Reply:
Sorry I think you misunderstood my request. I wouldn't normally push further on a logging change, but I think getting the right data will be helpful here.  

We're adding a configurable limit on total bytes processed. If someone is ever going to want to tune this, it'll be helpful to know how many total bytes are necessary to normalize. Even logging this over time will be useful to have a sense for throughput over time. So in addition to the list sizes, I think we should include the size we truncated at and the total size calculated by `calculatePlans`. That's why I was saying we might need to remove the `break`, since it'll require iterating all plans. I think it's still worth it since this isn't a super cpu intensive process.

Reply:
Fixed